### ISSUE TITLE
Help Regarding Hyper-V and Virtualization.Not a complain but i need smoll help :)

### ISSUE THREAD

**Issue by 0xReverse** [undergroundwires/privacy.sexy#363]
I casually opened hyper-v to test ubuntu server and as hyper-v is the best for bare metal virtualization.That mistake cost me my whole day.Browsing microsoft forums all day couldn't give me an answer.I was about to Rage Install Linux as Host OS until i remembered that i ran an important script which must have caused the malfunction.

I used the reverse  privacy-script.bat to undo changes and the VM ran smoothly after that.Like i don't know which part of the script is causing this.But if you use the script in any windows Hyper-V available version you can replicate the process like this:

After using the privacy script,you go right to hyper-v and test any OS to run as a VM.It will show the problems like screenshots given below:
![Screenshot 2024-05-12 095611_LI](https://github.com/undergroundwires/privacy.sexy/assets/169605882/0e6b0070-c95b-4047-9b90-76515e0ee721)
![Screenshot 2024-05-12 101031_LI](https://github.com/undergroundwires/privacy.sexy/assets/169605882/159a6b30-ae80-49b1-bae6-22c015ec49f0)

I would like to ask @undergroundwires as if there is any way to fix this issue by not reversing the whole script.Now the thing is i am not gonna sell my privacy just for some virtual machine nor i am wanna keep myself away from using virtual machine and ALSO I SUPPORT THE CONTRIBUTIONS OF THIS PROJECT from my heart.I would have donated if i had cash,too bad for me.

I have also read an old Issue from users that they've experienced windows sandbox getting broken after using the script and the owner documented it very well.

**Comment by 0xReverse** [comment]
You can test any other OS rather than ubuntu and it will show same results.Not limited to ubuntu only.

**Comment by undergroundwires** [comment]
Hi @0xReverse, do you have the script file you executed (you have it if you used the desktop app)? Or did you run any of the presets?

**Comment by 0xReverse** [comment]
Just Check Mark "Standard" Script and search Remote and see that if it has to do something with "Remote Management Or Remote Administration."

When i looked up i saw this option:
![20240513_232108](https://github.com/undergroundwires/privacy.sexy/assets/169605882/0f62f7f5-c603-4539-b7f8-9aef8c05b25f)

i dont know maybe hyper-v requires remote management to be enabled to stream even if it's local only?

**Comment by 0xReverse** [comment]
@undergroundwires I mean i also dont want the vulnearibilities of Remote management to be exploited as well so.....

**Comment by undergroundwires** [comment]
You can generate different combinations. I wonder about your configuration. Or did you just try "Standard", "Strict"?

I'd be happy to take action if you can figure out which script has caused this otherwise.

**Comment by 0xReverse** [comment]
> You can generate different combinations. I wonder about your configuration. Or did you just try "Standard", "Strict"?
> 
> I'd be happy to take action if you can figure out which script has caused this otherwise.

I used the "Standard" Script always.

**Comment by undergroundwires** [comment]
I could reproduce this on both Windows 11 Pro 23H2 and Windows 10 22H2 using Hyper-V and an Ubuntu VM.

I found that it's **Enable strong RSA key requirement** causes this, reverting it solves the issue.

This using Standard (binary search tree with bisect) and found out that `Enable strong RSA key requirement` causes this. Looking to its docs, we really want to keep them on "Standard":

![image](https://github.com/undergroundwires/privacy.sexy/assets/15555035/3bcbe830-8b8e-4c81-8980-76dd08c03c68)

But as it breaks Hyper-V connection, I believe we have to move it to Strict with a document about it breaking Hyper-V machine something like *(breaks Hyper-V VMs)*.

Thank you for the bug report and the context.

**Comment by 0xReverse** [comment]
> I could reproduce this on both Windows 11 Pro 23H2 and Windows 10 22H2 using Hyper-V and an Ubuntu VM.
> 
> I found that it's **Enable strong RSA key requirement** causes this, reverting it solves the issue.
> 
> This using Standard (binary search tree with bisect) and found out that `Enable strong RSA key requirement` causes this. Looking to its docs, we really want to keep them on "Standard":
> 
> ![image](https://private-user-images.githubusercontent.com/15555035/330390086-3bcbe830-8b8e-4c81-8980-76dd08c03c68.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTU2OTY5NzgsIm5iZiI6MTcxNTY5NjY3OCwicGF0aCI6Ii8xNTU1NTAzNS8zMzAzOTAwODYtM2JjYmU4MzAtOGI4ZS00YzgxLTg5ODAtNzZkZDA4YzAzYzY4LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA1MTQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNTE0VDE0MjQzOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTRlNDJmNmJjMjkxY2RmODYyZGY4MWIxOTA3YzMyMmMzZGQwYmNiNWNhMzhlMWEyM2VhY2Y2ZDA5YWIxZmI0NTMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.X4UgJqdpU3iiR2FMAB9n9wVACDu_o97zRWWkiPvZoN0)
> 
> But as it breaks Hyper-V connection, I believe we have to move it to Strict with a document about it breaking Hyper-V machine something like _(breaks Hyper-V VMs)_.
> 
> Thank you for the bug report and the context.

I am very sorry that i didnt tell you the Windows version the bug was being produced in.OMG i appreciate your work.But as it is a open source project you should take some rest 
.The way you are making commits for this project is appreciatable but you will lose your health.

Now should i wait for an Update in the script about it?Or Should i revert the "RSA Key" Option?

Is it only related to "RSA"? Kindly check also if it has to do something with Remote Authentication.Because i work with remote things very much and dont want to open a issue again on "SAME" thing.

**Comment by 0xReverse** [comment]
If it is possible to disable the "RSA Key Requirement" For distant remote connections and not for local than it would also make us more secure.

**Comment by 0xReverse** [comment]
@undergroundwires Should i only Revert "RSA Key Requirement" ?

Also does this script interfere with windows Update?

**Comment by undergroundwires** [comment]
Yes reverting only  "RSA Key Requirement" will solve the issue. The rest of the Standard is OK. It should not in anyway interfere with Windows Update, or its a bug.

**Comment by undergroundwires** [comment]
Yes reverting only  "RSA Key Requirement" will solve the issue. The rest of the Standard is OK. It should not in anyway interfere with Windows Update, or its a bug. Please report back if it solves for you or does not work.

We cannot granularly control this config, once it applies, it applies to all system connections.

**Comment by undergroundwires** [comment]
Yes reverting only  "RSA Key Requirement" will solve the issue.
Please report back if it solves for you or does not work.

The rest of the Standard is OK. It should not in anyway interfere with Windows Update, or its a bug. 

We cannot granularly control this config, once it applies, it applies to all system connections.

**Comment by undergroundwires** [comment]
Yes reverting only "RSA Key Requirement" will solve the issue.
Please report back if it solves for you or does not work.

The rest of the Standard is OK.
It should not in anyway interfere with Windows Update, or its a critical bug.

We cannot granularity control this config, once it applies, it applies to all system connections.

My further tests show that Docker, wsl, and Windows Sandbox work fine. This impacts only Hyper-V VM connection, connections to other virtualization-based solutions work fine.

**Comment by undergroundwires** [comment]
[0.13.4](https://github.com/undergroundwires/privacy.sexy/releases/tag/0.13.4) is just released 🎉. Actions taken:

- Add documentation to describe potential disruptions caused by stronger RSA key requirements.
- Move RSA key script from 'Standard' to 'Strict' due to its impact on Hyper-V VMs.
  
Thank you @0xReverse.

Please create new issues for further ideas/problems.

**Comment by zekesulastin** [comment]
dio Model 100 - Firmware version 0.92.6+116
